pages/scripts/exporthardcopy.js

```diff
@@ -0,0 +1,271 @@
+    });
+  }
+
+  if(unit.length == 0){
+    console.log('No Units Selected');
+  }else{
+    console.log(unit);
+  }
+
+
+  LighthouseJob.get_json(unit, host, start, end, function(jobs) {
+
+    $.each(jobs.Results , function(k,v){
+      var thisJob = jobTemplate;
+      _.each(selectedcolumns, function(key){
+        // Raw Value
+        var rawValue = key.split('.').reduce(function(obj,i) {return obj[i]}, v);
+        // Special Cases
+        switch(key){
+          case 'JobReceived':
+            var rawDate = new Date(v.JobReceived);
+            rawDate = new Date(rawDate.getTime() + ( rawDate.getTimezoneOffset() * 60000 ));
+            rawValue = (''+rawDate).split( ' GMT' )[0];
+            break;
+          case 'Tags':
+            rawValue = v.Tags.map(function(v){return v.Name}).join(' , ');
+            break;
+          case 'JobPriorityType.Name':
+            rawValue = ( rawValue == 'General' ? '' : ' ('+rawValue+')' );
+            break;
+          case 'PermissionToEnterPremises':
+            rawValue = ( rawValue ? 'P' : '<strong>NO</strong> P' )+'ermission granted to access premises if resident is absent';
+            break;
+        }
+        if( rawValue == null )
+          rawValue = '';
+        thisJob = thisJob.replace( '++' + key + '++' , rawValue );
+      });
+      thisJob = thisJob.replace( /\+\+[^\+]+\+\+/g , '<em>Unknown or Unavailable</em>' );
+      $('#results').append(thisJob);
+    });
+
+    //progressBar && progressBar.setValue(1);
+    //progressBar && progressBar.close();
+    //downloadCSV("LighthouseExport.csv", beacon_jobs, selectedcolumns);
+
+    //progressBar && progressBar.close();
+
+    $('#loadinner p').text('Loaded');
+
+    window.print();
+
+  },function(val,total){
+    progressBar && progressBar.setValue(val/total);
+  });
+}
+
+function convertArrayOfObjectsToCSV(data, selectedcolumns) {  
+
+  var result, ctr, keys, columnDelimiter, lineDelimiter, data;
+
+  if (data == null || !data.length) {
+    return null;
+  }
+
+  //console.log('data', data);
+
+  var rows = [];
+  var delimCellL = '"=""';
+  var delimCellR = '"""';
+  var delimCol   = delimCellR + ',' + delimCellL;
+  var delimLine  = '\n';
+
+  // Getting Column Headings
+  var fieldKeys = fieldLabels = [];
+  _.each(lighthouse_fieldArray, function(fields, section){
+    _.each(fields, function(label, key){
+      fieldKeys.push(key);
+      fieldLabels.push(label);
+    });
+  });
+  selectedcolumns.map(function(v, k){
+    return fieldLabels[fieldKeys.indexOf(v)];
+  })
+
+  rows.push( delimCellL + selectedcolumns.join( delimCol ) + delimCellR );
+
+  data.forEach(function(item){
+    rows.push( delimCellL + item.join( delimCol ) + delimCellR );
+  });
+
+  return rows.join( delimLine );
+}
+
+
+function downloadCSV(file, dataIn, keyIn) {  
+  var csv = convertArrayOfObjectsToCSV(dataIn, keyIn);
+  if (csv == null)
+    return;
+
+var saveData = (function () {
+    var a = document.createElement("a");
```

Comment:
This is actually cruft - old functions from the copy-paste which I didn't end up using. Still need to do some housekeeping.